ci/ci.yml

```diff
@@ -232,262 +233,71 @@ Resources:
-    Type: "AWS::Lambda::Function"
-    Properties:
-      Code:
-        S3Bucket:
-          Fn::ImportValue:
-            !Sub ${InfrastructureStorageStackName}-InfrastructureSupportBucket
-        S3Key: "ci/ci-status-notification-handler.zip"
-      Description: >
-        Handles status notifications for CI builds
-      Environment:
-        Variables:
-          SLACK_MESSAGE_RELAY_TOPIC_ARN:
-            Fn::ImportValue:
-                !Sub ${InfrastructureNotificationsStackName}-SlackMessageRelaySnsTopicArn
-      Handler: index.handler
-      MemorySize: 128
-      Role: !GetAtt CiStatusNotificationHandlerIamRole.Arn
-      Runtime: nodejs12.x
-      Tags:
-        - Key: Project
-          Value: Infrastructure
-        - Key: "prx:cloudformation:stack-name"
-          Value: !Ref AWS::StackName
-        - Key: "prx:cloudformation:stack-id"
-          Value: !Ref AWS::StackId
-      Timeout: 10
-  CiStatusNotificationPermission:
-    Type: "AWS::Lambda::Permission"
-    Properties:
-      Action: "lambda:InvokeFunction"
-      FunctionName: !Ref CiStatusNotificationHandlerLambdaFunction
-      Principal: sns.amazonaws.com
-      SourceArn: !Ref CiStatusSnsTopic
-  CiStatusNotificationSubscription:
-    Type: "AWS::SNS::Subscription"
-    Properties:
-      Endpoint: !GetAtt CiStatusNotificationHandlerLambdaFunction.Arn
-      Protocol: lambda
-      TopicArn: !Ref CiStatusSnsTopic
-  CiStatusNotificationHandlerErrorAlarm:
-    Type: "AWS::CloudWatch::Alarm"
-    Properties:
-      ActionsEnabled: true
-      AlarmName: "[Infra][CI][Lambda][Errors] CI Status Handler"
-      AlarmActions:
-        - Fn::ImportValue:
-            !Sub ${InfrastructureNotificationsStackName}-OpsWarnMessagesSnsTopicArn
-      InsufficientDataActions:
-        - Fn::ImportValue:
-            !Sub ${InfrastructureNotificationsStackName}-OpsWarnMessagesSnsTopicArn
-      OKActions:
-        - Fn::ImportValue:
-            !Sub ${InfrastructureNotificationsStackName}-OpsWarnMessagesSnsTopicArn
-      AlarmDescription: >
-        The error rate on the CI status notifications handler lambda has
-        exceeded 0 in the last minute.
-      ComparisonOperator: GreaterThanThreshold
-      EvaluationPeriods: 1
-      MetricName: Errors
-      Namespace: AWS/Lambda
-      Period: 60
-      Statistic: Sum
-      Threshold: 1
-      TreatMissingData: notBreaching
-      Dimensions:
-        - Name: FunctionName
-          Value: !Ref CiStatusNotificationHandlerLambdaFunction
-  CiStatusNotificationHandlerLambdaFunctionRetentionPolicy:
-    Type: "Custom::LogGroupRetentionPolicy"
-    Properties:
-      ServiceToken: !Ref LogGroupRetentionPolicyFunctionArn
-      LogGroupName: !Sub "/aws/lambda/${CiStatusNotificationHandlerLambdaFunction}"
-      RetentionInDays: 5
+        Project: Infrastructure
+        "prx:cloudformation:stack-name": !Ref AWS::StackName
+        "prx:cloudformation:stack-id": !Ref AWS::StackId
+  # HttpApiDomainNameRecordSetGroup:
```

Comment:
Why commented out?  Just making sure the old recordset gets completely deleted first?

Reply:
Because of the way SAM transforms these SAM resources into actual Cfn resources behind the scenes, sometimes there are cases where you can't reference the same outputs you would if it were done in pure Cfn. It's kind of a limitation, and if this were a more important host name I would have just done it less magically, but for this I just switched the GitHub webhook to the domain provided by API Gateway.